accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+```csharp
+public static double GetStandardDeviation(this IEnumerable<int> values)
+{
+    double standardDeviation = 0;
+    int[] enumerable = values as int[] ?? values.ToArray();
+    int count = enumerable.Count();
+    if (count > 1)
+    {
+        double avg = enumerable.Average();
+        double sum = enumerable.Sum(d => (d - avg) * (d - avg));
+        standardDeviation = Math.Sqrt(sum / count);
+    }
+    return standardDeviation;
+}
+```
+
+Jeff copies this code into his project and is now able to calculate the standard deviation for a set of integers. However, Jeff also wants to support other types such as `float`, `double`, or even user-defined inputs. Jeff has realized that creating a reusable and easily maintainable implementation would allow producing a NuGet package that other developers seeking the same could depend upon.
+
+After thinking over the possible implementations, Jeff realizes that the only reliable way to implement this in .NET 5 is to define an interface with instance methods that users must then implement on their own types or to hardcode a list of types in their own library. They do not believe this to be as reusable or maintainable as hoped.
+
+Jeff learns that a future version of .NET will offer a new feature called static abstracts in interfaces. Such a feature will have the .NET Libraries expose a set of interfaces that allow Jeff to implement the desired method without requiring users types to take a dependency on types they declared.
+
+The .NET Libraries are exposing interfaces such as:
+```csharp
+public interface IAddable<TSelf>
+    where TSelf : IAddable<TSelf>
+{
+    static abstract TSelf operator +(TSelf left, TSelf right);
+}
+
+public interface ISubtractable<TSelf>
+    where TSelf : ISubtractable<TSelf>
+{
+    static abstract TSelf operator -(TSelf left, TSelf right);
+}
+
+public interface IMultipliable<TSelf>
+    where TSelf : IMultipliable<TSelf>
+{
+    static abstract TSelf operator -(TSelf left, TSelf right);
+}
+
+public interface IDivisable<TSelf>
+    where TSelf : IDivisable<TSelf>
+{
+    static abstract TSelf operator /(TSelf left, TSelf right);
+}
+
+public interface INumber<TSelf>
+    : IAddable<TSelf>
+    : ISubtractable<TSelf>
+    : IMultipliable<TSelf>
+    : IDivisable<TSelf>
+    where TSelf : INumber<TSelf>
+{
+}
+
+// New methods on System.Linq.Enumerable for `Sum<T>` and `Average<T>`, `where T : INumber<T>`
+```
+
+Jeff can now implement their method such as:
+```csharp
+public static double GetStandardDeviation<T>(this IEnumerable<T> values)
+    where T : INumber<T>
+{
+    double standardDeviation = 0;
+    T[] enumerable = values as T[] ?? values.ToArray();
+    int count = enumerable.Count();
+    if (count > 1)
+    {
+        double avg = enumerable.Average();
+        double sum = enumerable.Sum(d => (d - avg) * (d - avg));
```

Comment:
I forgot to put in the conversion to double here. It would be:
```suggestion
        double sum = enumerable.Sum(d => ((double)d - avg) * ((double)d - avg));
```

(or `double sum = enumerable.Sum(d => (double.CreateFrom(d) - avg) * (double.CreateFrom(d) - avg));`)